src/Database/Redis/Connection.hs

```diff
@@ -275,12 +275,12 @@ shardMapFromClusterSlotsResponse ClusterSlotsResponse{..} = ShardMap <$> foldr m
         in
             Cluster.Node clusterSlotsNodeID role hostname (toEnum clusterSlotsNodePort)
 
-refreshShardMap :: ConnectInfo -> Cluster.Connection ->  IO ShardMap
-refreshShardMap connectInfo@ConnInfo{connectMaxConnections,connectMaxIdleTime} (Cluster.Connection shardNodeVar _ _) = do
+refreshShardMap :: ConnectInfo -> Cluster.Connection -> Maybe Cluster.NodeConnection -> IO (ShardMap, NodeConnectionMap)
+refreshShardMap connectInfo@ConnInfo{connectMaxConnections,connectMaxIdleTime} (Cluster.Connection shardNodeVar _ _) nodeConn = do
     modifyMVar shardNodeVar $ \(_, oldNodeConnMap) -> do
-        newShardMap <- refreshShardMapWithNodeConn (HM.elems oldNodeConnMap)
+        newShardMap <- refreshShardMapWithNodeConn nodeConn (HM.elems oldNodeConnMap)
         newNodeConnMap <- updateNodeConnections newShardMap oldNodeConnMap        
-        return ((newShardMap, newNodeConnMap), newShardMap)
+        return ((newShardMap, newNodeConnMap), (newShardMap, newNodeConnMap))
```

Comment:
We can extract the NodeConnectionMap from ShardMap datastructure itself ?

Reply:
Oh, got it. So given the current structure, we have to satisfy the refreshAction signature as (Maybe NodeConnection -> IO (ShardMap, NodeConnectionMap))